code/modules/unit_tests/find_reference_sanity.dm

```diff
@@ -67,3 +67,25 @@
 	testbed.test_list.Cut()
 	to_find_assoc.Cut()
 	testbed.test_assoc_list.Cut()
+
+	//Calm before the storm
+	testbed.test_assoc_list = list(null = victim)
+
+	victim.DoSearchVar(testbed, "Fourth Run", search_time = 5)
+	TEST_ASSERT(testbed.test_assoc_list, "The ref-tracking tool failed to find a null key'd assoc list entry")
+	victim.found_refs.Cut()
+	testbed.test_assoc_list.Cut()
+
+	//Let's do some more complex assoc list investigation
+	var/list/to_find_in_key = list(victim)
```

Comment:
Make this a separate test.

Reply:
why? it's the same concept as the rest of the unit test, several different testcases all of which test the stability of DoSearchVar(). Is it a bloat thing?

Reply:
Unit tests should, ideally, be testing one behavior each.

Reply:
Should I give each one its own file? or keep them in the same place but under different headers

Reply:
I split them, will this do?